### PR TITLE
Add dead letter queue support

### PR DESCRIPTION
**Pull request by agoncharov-reef**
nuff said

### CODE REVIEW

File: {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/{{cookiecutter.django_default_app_name}}/metrics.py [hunk 1]
```diff
@@ -33,4 +37,14 @@ def metrics_view(request):
         )
     else:
         return ExportToDjangoView(request)
+
+
+num_tasks_in_queue = {}
+for queue in settings.CELERY_TASK_QUEUES:
+    gauge = prometheus_client.Gauge(
```
Comment (ppolewicz) [inline {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/{{cookiecutter.django_default_app_name}}/metrics.py:44]: ❤️

File: {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/{{cookiecutter.django_default_app_name}}/tasks.py [hunk 2]
```diff
@@ -1,13 +1,40 @@
...
+    Once the reason of tasks failure is fixed, the task can be re-processed
+    by running a worker on dead letter queue:
+
+        celery -A mrc.celery worker -Q dead_letter
+
+    If tasks fails again, it will be put back to dead letter queue. This can
+    potentially create an infinite loop, so make sure to fix the reason of
+    failure before re-processing the task, or stop the worker after a few
+    retries. The process of re-processing tasks is supposed to be manual,
+    thus there are no complicated mechanisms to prevent infinite loops.
```
Comment (ppolewicz) [inline {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/{{cookiecutter.django_default_app_name}}/tasks.py:25]: Shouldn't we add a dead dead letter queue for this and then the fixing processor would be configured to pick stuff up from dead_letter and run it but on failure put in dead_dead_letter, then if this happens you can fix the code in such a way that it's not sending anything to dead_dead, run through the whole dead and finally run like a special task which takes everything from dead_dead and moves to dead or runs on dead_dead and puts to dead? I doubt we'll always be perfect and there isn't anything that would warn us about the infinite loop, so putting in a process to handle it properly seems practical to me. It's a good moment to think about how to handle it.

Also this sounds like something that should be described somewhere in the documentation of our framework, not just in this docstring here.

File: {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/{{cookiecutter.django_default_app_name}}/tasks.py [hunk 3]
```diff
@@ -1,13 +1,40 @@
...
+    """
+    if task.app.conf.task_always_eager:
+        return  # do not run failed task again in eager mode
+
+    logger.debug(
```
Comment (ppolewicz) [inline {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/{{cookiecutter.django_default_app_name}}/tasks.py:30]:
```suggestion
    logger.warning(
```

File: {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/{{cookiecutter.django_default_app_name}}/tasks.py (hunk 2 above)
Comment (agoncharov-reef) [reply {{cookiecutter.repostory_name}}/app/src/{{cookiecutter.django_project_name}}/{{cookiecutter.django_default_app_name}}/tasks.py]: I just added a command to move tasks from one queue to another. No infinite loops involved.

https://github.com/reef-technologies/cookiecutter-rt-django/commit/322fc5e89692b00a158da25ee3b9402ff9edacde

File: {{cookiecutter.repostory_name}}/README.md [hunk 4]
```diff
@@ -144,6 +144,28 @@ If an SSO provider supports the OIDC protocol, it can be set up as a generic OID
 2. Allauth "social users" are just an extension to regular django users. When someone logs in via allauth, a django user model will also be created for them.
 3. A "profile" page is available at `/accounts/`
 
+{% endif %}
+
+{% if cookiecutter.use_celery == 'y' %}
+# Background tasks with Celery
```
Comment (mjurbanski-reef) [inline {{cookiecutter.repostory_name}}/README.md:150]: please add `<details>` tags so this is collapsible

**Pull request merged**